lib/spack/spack/test/cmd/env.py

```diff
@@ -984,12 +984,35 @@ def test_bad_env_yaml_format(tmpdir):
     - mpileaks
 """
         )
+    f.close()
 
     with tmpdir.as_cwd():
-        with pytest.raises(spack.config.ConfigFormatError) as e:
-            env("create", "test", "./spack.yaml")
-        assert "spack.yaml:2" in str(e)
-        assert "'spacks' was unexpected" in str(e)
+        with pytest.raises(spack.main.SpackCommandError) as e:
+            output = env("create", "test", "./spack.yaml")
+            assert "'spacks' was unexpected" in str(output)
+            assert "exited with code 1: env('create', 'test', './spack.yaml')\"" in str(e)
+
+        assert "test" not in env("list")
+
+    badenv = "badenv"
+    env("create", badenv)
+    tmpdir = spack.environment.environment.environment_dir_from_name(badenv, exists_ok=True)
+    filename = os.path.join(tmpdir, "spack.yaml")
+    with open(filename, "w") as f:
+        f.write(
+            """\
+spack:
+  spacks:
+    - mpileaks
```

Comment:
FYI. You could even change this up to get a different error, e.g., `spacks` -> `specs` and `- mpileaks` to simply `mpileaks`.